tardis/resources/drone.py

```diff
@@ -93,6 +93,12 @@ def site_agent(self) -> SiteAgent:
         return self._site_agent
 
     async def run(self):
+        if not self.state:
```

Comment:
While I think no state will ever be false'y, let's still better be robust here and make a precise check:
```suggestion
        if self.state is None:
```